### PR TITLE
Adds SemVer support to papiea packages

### LINKED ISSUE #484: Set up proper Papiea semantic versioning
1) Papiea components should follow the same semantic versioning.
2) (Optionally) set up public repos: docker hub, npm, pip
3) Versioning should be meaningful, version should REALLY comply to our progress

### PR DESCRIPTION
**Pull request by Emulebest**
resolves #484

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -106,6 +104,6 @@ workflows:
       - release:
           requires:
             - build
-          filters:
-            branches:
-              only: master
+#          filters:
+#            branches:
+#              only: master
```
Comment (shlomi-nx) [inline .circleci/config.yml:109]: Why are these removed?

File: README.md [hunk 2]
```diff
@@ -13,9 +13,9 @@ Please see [Papiea's design document](https://nutanix.github.io/papiea/Papiea-de
 
 | Component  | Version |
 | ------------- | ------------- |
-  | Engine (docker) | 1411 |
-  | Client/SDK (typescript)  | 1.0.108 |
-  | Client/SDK (python)  | dd4d35ee9cbd21d24abd7b20e0e12c6d9a35233b |
+  | Engine (docker) | nutanix-docker.jfrog.io/papiea:1.0.210_1446 |
+  | Client/SDK (typescript)  | 1.0.210+1446 |
```
Comment (shlomi-nx) [inline README.md:17]: I may have missed some of the discussion on this, but is there a way to settle on one character to split version and its build number?

File: README.md [hunk 3]
```diff
@@ -13,9 +13,9 @@ Please see [Papiea's design document](https://nutanix.github.io/papiea/Papiea-de
...
+  | Client/SDK (python)  | 74d01a0a3fb66c5a38cb866ed800ab04ebc302e6 |
```
Comment (shlomi-nx) [inline README.md:18]: I suppose we will get to this at some point, right?

File: papiea-backend-utils/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "papiea-backend-utils",
-  "version": "1.0.108",
+  "version": "1.0.217+9999",
```
Comment (shlomi-nx) [inline papiea-backend-utils/package.json:3]: Why are we committing this `9999` version?

**Review by shlomi-nx — COMMENTED** [review]
Looks great. A few comments:

File: .circleci/config.yml (hunk 1 above)
Comment (Emulebest) [reply .circleci/config.yml:109]: Debugging not on master :)

File: README.md (hunk 2 above)
Comment (Emulebest) [reply README.md:17]: @joshua-berry-ntnx told that people might still get what the `_` stands for, as for semver specs, it only allows `+` which is prohibited by the docker hub tagging

File: README.md (hunk 3 above)
Comment (Emulebest) [reply README.md:18]: If you mean a single version, we are almost at it!

File: papiea-backend-utils/package.json (hunk 4 above)
Comment (Emulebest) [reply papiea-backend-utils/package.json:3]: This is a commit to retrigger CI run, `9999` is a manually put CI number in the commit. While running CircleCI it gets this from env variable. Take a look at this for example https://github.com/nutanix/papiea/pull/497/commits/f30a8b5b7da11760c236d0349e3e0e0c8290629e

File: README.md (hunk 2 above)
Comment (joshua-berry-ntnx) [reply README.md]: To add onto this, the issue is that semver expects `+` to separate build tag and version number, but Docker does not support `+` characters in tags. So we settled on this as a way to make sure the Docker version still contains all the information while still trying to comply with semver as best we can.
